simpleviewinc/tap-events-force
# Zen 626 mobile state label

**Ticket**: https://jira.simpleviewtools.com/browse/ZEN-626

## Context

* New UI requirements dictate that state labels appear on the top right of the desktop sessions and also appear for mobile sessions
* Please refer new UI expectations here : https://www.figma.com/file/aLesACdNG3CvInUVuFm4hO/Session-selection---final?node-id=0%3A1

## Goal

* For Desktop : Change state button to appear on the top right rather that bottom right. 
* For Desktop : Ensure price (dollar amount / free etc) that originally existed on the top right is no longer displayed
* For Desktop : When a session is clicked/expanded , the state button now appears on the top right rather bottom right
* For Mobile : Ensure when collapsed a session shows non interactive state label except 'select' and 'read ' on the top right
* For Mobile : Ensure when expanded the state label disappears and existing functionality where state button appears at the bottom of the session remains

## Updates

* keg-hub/taps/tap-events-force/src/components/grid/gridTileContent.js
  * For desktop collapsed view: Deleted the display of price and added the button as a compound element in row 1
* keg-hub/taps/tap-events-force/src/theme/components/grid/gridTileContent.js
  * For desktop collapsed view: Removed styles that were not in use
* keg-hub/taps/tap-events-force/src/components/modals/sessionDetailsModal.js
  *  For desktop expanded view: Deleted the display of price and added the button as a compound element in row 1
* keg-hub/taps/tap-events-force/src/theme/components/modals/sessionDetailsModal.js
  * For desktop collapsed view: Modified styles as required
* keg-hub/taps/tap-events-force/src/components/grid/gridRowContent.js
  * For mobile collapsed and expanded view ensure state labels other that select and read-only appear on top right and disappear when expanded
* keg-hub/taps/tap-events-force/src/theme/components/grid/gridRowContent.js
  * For mobile expanded and collapsed session view , modified styles
* keg-hub/taps/tap-events-force/src/theme/components/grid/gridItem.js
  * Ensuring compound row containing time and state label appear as desired
* keg-hub/taps/tap-events-force/src/components/labels/stateLabel.js
  * Adding new component for the right side of row 1, that contains state text and icon if applicable

## Testing

* SetUp
  * Run the command => keg evf pack run evf:zen-626-mobilestatelabel
  * Navigate to http://evf-zen-626-mobilestatelabel.local.keghub.io/

* Test 1 : Desktop collapsed session test
  * Launch events force from the docker image above and navigate to main page indicated in setup
  * Ensure for the session named 'Zerista session day 3' at bottom of page , price is no longer visible and select button is at the top right of the session tile
  * Ensure for session titled 'session at same time other session with presenters and labels' , select button is present at top right of session tile and FREE is no longer displayed
  * Ensure for session titled 'Session on day 2 - limited capacity', the 'on waiting list:1' button is present at the top right
  * Note : since the use case only required changing the placement of the state button (and removing price), testing for other states is not required and the above tests should be sufficient . If however you would like to test other states , please select appropriate data and do so.

* Test 2 : Desktop expanded session test (session modal test)
  * Launch events force from the docker image above and navigate to main page indicated in setup
  * Click on 'Zerista session day 3' session . It should open up a modal. Ensure price is no longer visible and select button is at the top right
  * Click on 'session at same time other session with presenters and labels' . It should open up a modal. Ensure FREE is no longer visible and select button is at the top right
  * Click on 'Session on day 2 - limited capacity' . Ensure the 'on waiting list:1' button is present at the top right

* Test 3 : Mobile session expanded/collapsed test
  * Launch events force from the docker image above and navigate to main page indicated in setup
  * Right click on page and click inspect
  * In the inspector window that opens up , select mobile view on the top left : 'toggle device toolbar'. It should open up the app in a view similar to mobile
  * Ensure for session 'session at same time other session with presenters and labels', there is no state label on top right for the collapsed session tile. When you click it , thereby expanding it , there should be a select button the the bottom
  * Ensure for session titled 'Session on day 2 - limited capacity', the 'On Waiting list' label WITHOUT the number shows up on the top right . It should be in Caps and styled according to figma template
  * When clicked / expanded the state label should disappear and existing functionality with button showing 'On waiting list : 1 ' should become visible

* Test 4 : Mobile session expanded/collapsed test for state 'SELECTED'
  * Launch events force from the docker image above and navigate to main page indicated in setup
  * Right click on page and click inspect
  * In the inspector window that opens up , select mobile view on the top left : 'toggle device toolbar'. It should open up the app in a view similar to mobile
  * Click on 'session at same time other session with presenters and labels' . Its the second session from the top . 
  * Click on the 'select' button and then click on 'book selected'. It should process and say 'Selected 4'
  * Minimize the session again
  * Ensure that the 'SELECTED' label appears on the top right with a checkmark icon next to it

* Test 5 : Mobile session expanded/collapsed test for state 'FULLY BOOKED'
  * Launch events force from the docker image above and navigate to main page indicated in setup
  * Right click on page and click inspect
  * In the inspector window that opens up , select mobile view on the top left : 'toggle device toolbar'. It should open up the app in a view similar to mobile
  * Click on 'Test Data-Json' and select fullybooked-single-stopped. Apply changes
  * Ensure that the 'FULLY BOOKED' label appears on the top right in a disabled color. 
  * When you click on the session and it expands , the label should disappear

* Test 6 : Mobile session expanded/collapsed test for specific states
  * Launch events force from the docker image above and navigate to main page indicated in setup
  * Right click on page and click inspect
  * In the inspector window that opens up , select mobile view on the top left : 'toggle device toolbar'. It should open up the app in a view similar to mobile
  * States SELECT and READ ONLY should not appear as labels for collapsed session. Only states SELECTED, FULLY BOOKED, WAITING LIST and ON WAITING LIST can appear
